uber/templates/preregistration/index.html

```diff
@@ -68,6 +68,7 @@
             </ul>
         </td>
         <td>${{ attendee.total_cost }}</td>
+        <td><form action="redeem"><input type="text" name="edit_id" value="{{ attendee.id }}" hidden> <input placeholder="Enter Promo Code" type="text" name="promo"><input type="submit" value="Submit"></form></td>
```

Comment:
You could hide it even more by having it only 'enterable' through a special URL, e.g., `form?promo_code=BLAH`. In the form function, you could do something like:
```
promo_code = params['promo_code'] if 'promo_code' in params else None
# Other stuff
return {
  #blahblah all your other vars
  promo_code: promo_code
}
```
Then, you could make a hidden input field with `promo_code`. This is all off the top of my head so it may not work exactly as I've written it, though.

The idea being that the function would grab the promo_code from the GET parameters and turn it into a POST parameter.